crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1152,6 +1158,56 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
             )));
         }
 
+        // Check whether this version covers all supported platforms.
+        if !dist.implied_markers().is_true() {
```

Comment:
We only go down this path for distributions that lack a source distribution (which is comparatively quite rare).